picard/ui/sortablecheckboxlist.py

```diff
@@ -0,0 +1,171 @@
+        up = layout.itemAtPosition(row, self._BUTTON_UP).widget()
+        down = layout.itemAtPosition(row, self._BUTTON_DOWN).widget()
+        checkbox.stateChanged.connect(partial(self.checkbox_toggled, row))
+        up.clicked.connect(partial(self.move_button_clicked, row, up=True))
+        down.clicked.connect(partial(self.move_button_clicked, row, up=False))
+
+    def moveItem(self, from_row, to_row):
+        to_row = to_row % len(self.__items)
+        self.__items[to_row], self.__items[from_row] = \
+            self.__items[from_row], self.__items[to_row]
+        self.updateRow(to_row)
+        self.updateRow(from_row)
+        self._emit_changed()
+
+    def checkbox_toggled(self, row, state):
+        self.__items[row].checked = (state == QtCore.Qt.Checked)
+        self._emit_changed()
+
+    def move_button_clicked(self, row, up):
+        if up:
+            to = row - 1
+        else:
+            to = row + 1
+        self.moveItem(row, to)
+
+    def updateRow(self, row):
+        self.__no_emit = True
+        item = self.__items[row]
+        layout = self.layout()
+        checkbox = layout.itemAtPosition(row, self._CHECKBOX_POS).widget()
+        checkbox.setText(item.text)
+        checkbox.setChecked(item.checked)
+        self.__no_emit = False
+
+    def addItem(self, item):
+        self.__items.append(item)
+        row = len(self.__items) - 1
+        layout = self.layout()
+        layout.addWidget(QtGui.QCheckBox(), row, self._CHECKBOX_POS)
+        self.updateRow(row)
+        up_button = QtGui.QToolButton()
+        up_button.setArrowType(QtCore.Qt.UpArrow)
+        up_button.setMaximumSize(QtCore.QSize(16, 16))
+        down_button = QtGui.QToolButton()
+        down_button.setArrowType(QtCore.Qt.DownArrow)
+        down_button.setMaximumSize(QtCore.QSize(16, 16))
+        layout.addWidget(up_button, row, self._BUTTON_UP)
+        layout.addWidget(down_button, row, self._BUTTON_DOWN)
+        self.setSignals(row)
+
+    def _emit_changed(self):
+        if not self.__no_emit:
+            self.changed.emit(self.__items)
+
+
+class SortableCheckboxListItem:
+
+    def __init__(self, text=u'', checked=False, data=None):
+        self._checked = checked
+        self._text = text
+        self._data = data
+
+    @property
+    def text(self):
+        return self._text
+
+    def setText(self, text):
+        self._text = text
+
+    @property
+    def checked(self):
+        return self._checked
+
+    def setChecked(self, state):
+        self._checked = state
+
+    @property
+    def data(self):
+        return self._data
+
+    def setData(self, data):
+        self._data = data
+
+    def __repr__(self):
+        params = []
+        params.append('text=' + repr(self.text))
+        params.append('checked=' + repr(self.checked))
+        if self.data is not None:
+            params.append('data=' + repr(self.data))
+        return "%s(%s)" % (self.__class__.__name__, ", ".join(params))
+
+
+def main():
```

Comment:
Removed in https://github.com/zas/picard/commit/f82056007cd3038f432c8450930ce80c95306e3b